hub/api/tests/test_api.py

```diff
@@ -0,0 +1,75 @@
+import os, shutil
+import numpy as np
+from hub.api.dataset import Dataset
+from hub.core.chunk_engine.read import read_dataset_meta, read_tensor_meta
+import pytest
+
+
+@pytest.fixture(autouse=True)
+def clear_test_files():
+    shutil.rmtree("/tmp/hub-test/", ignore_errors=True)
+
+
+def test_create_empty_memory_dataset():
```

Comment:
these tests will need to support all storage providers & since `pytest-options` is handling the complexity of managing these fixtures, no worries here.